typing/typecore.ml

```diff
@@ -390,16 +395,18 @@ let is_principal ty =
 (* Returns [Some ty_elt] if the expected type can be used for type-directed
    disambiguation of an array literal. *)
```

Comment:
The comment is now stale. What is going on is slightly subtle, I would not mind a dedicated record type to represent the information.